ValveResourceFormat/Resource/ResourceTypes/KV3.cs

```diff
@@ -0,0 +1,592 @@
+/*
+ * KV3Reader class.
+ * This class reads in valve KV3 files and stores them in its datastructure.
+ *  
+ * Interface:
+ *  KVFile file = KV3Reader.ParseKVFile( fileName );
+ *  String fileString = file.Serialize();
+ *  
+ * TODO:
+ *  - Test some more and find the bugs.
+ *  - Revisit state machine if bugs require it.
+ *  - Improve KVFile interface.
+ *  
+ * Author: Perry - https://github.com/Perryvw
+ * 
+ */
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.IO;
+using System.Globalization;
+
+namespace ValveKV
+{
+    public static class KV3Reader
```

Comment:
Agreed. This is not thread safe and no way to make it thread safe, unless you resign to never doing concurrent KV3 serialization.